### PR TITLE
Ask va api refactor translator

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
## Summary
Enhance Translation Support for Veteran and Dependent Relationships
This PR introduces key improvements and minor fixes to enhance translation accuracy across the application.

1. New dictionary.yml file:
	• Added dictionary.yml to support translations for veteran_relationship and dependent_relationship.
2. Updates to en.yml and optionset_mock_data:
	• Synchronized en.yml and optionset_mock_data with new translation entries.
3. Enhancements in Translator.rb:
	• Updated Translator.rb to accurately translate all optionsets, ensuring alignment with the latest dictionary.yml.
4. Spec and Submitter Profile Fixes:
	• Minor adjustments in specs and submitter_profile to resolve inconsistencies and improve reliability.

Notes:

These changes aim to standardize relationship translations and improve translation handling within optionsets, ensuring a more robust translation framework.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/ask-va/issues/1463

## Testing done

- [x] *New code is covered by unit tests*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

**Review by hemeshvpatel — APPROVED** [review]
Lgtm!

**Review by hemeshvpatel — DISMISSED** [review]
lgtm!

**Review by hemeshvpatel — APPROVED** [review]
Lgtm

File: modules/ask_va_api/app/lib/ask_va_api/inquiries/payload_builder/inquiry_details.rb [hunk 1]
```diff
@@ -87,11 +87,11 @@ def handle_others_inquiry(inquiry_params, inquiry_details)
...
-             inquiry_params[:their_relationship_to_veteran]
+             inquiry_params[:more_about_your_relationship_to_veteran]
             build_inquiry_details(
               inquiry_details: inquiry_details,
               inquiry_about: 'On Behalf of a Veteran',
-              veteran_relationship: inquiry_params[:their_relationship_to_veteran]
+              veteran_relationship: inquiry_params[:more_about_your_relationship_to_veteran]
             )
           elsif inquiry_params[:is_question_about_veteran_or_someone_else] == 'Someone else' &&
                 inquiry_params[:their_relationship_to_veteran]
```
Comment (RachalCassity) [inline modules/ask_va_api/app/lib/ask_va_api/inquiries/payload_builder/inquiry_details.rb:97]: Should `their_relationship_to_veteran` be used here?

Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/app/lib/ask_va_api/inquiries/payload_builder/inquiry_details.rb:97]: No with the flow of the application. I'll be getting `more_about_your_relationship_to_veteran` and not `their`

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**